core/src/main/scala/spire/math/Rational.scala

```diff
@@ -554,29 +556,38 @@ private[math] object LongRationals extends Rationals[Long] {
       else LongRational(-n, d)
 
     def +(r: Rational): Rational = r match {
-      case r: LongRationals.LongRational =>
+      case r: LongRational =>
         val dgcd: Long = spire.math.gcd(d, r.d)
         if (dgcd == 1L) {
-
-          val num = SafeLong(n) * r.d + SafeLong(r.n) * d
-          val den = SafeLong(d) * r.d
-          Rational(num, den)
+          Checked.tryOrReturn[Rational] {
+            build(n * r.d + r.n * d, d * r.d)
+          } {
+            Rational(SafeLong(n) * r.d + SafeLong(r.n) * d, SafeLong(d) * r.d)
+          }
 
         } else {
 
           val lden: Long = d / dgcd
           val rden: Long = r.d / dgcd
-          val num: SafeLong = SafeLong(n) * rden + SafeLong(r.n) * lden
+          Checked.tryOrReturn {
+            val num: Long = n * rden + r.n * lden
 
-          val ngcd: Long = num match {
-            case SafeLongLong(x) => spire.math.gcd(x, dgcd)
-            case SafeLongBigInt(x) => spire.math.gcd(dgcd, (x % dgcd).toLong)
-          }
+            val ngcd: Long = spire.math.gcd(num, dgcd)
 
-          if (ngcd == 1L)
-            Rational(num, SafeLong(lden) * r.d)
-          else
-            Rational(num / ngcd, SafeLong(lden) * (r.d / ngcd))
+            if (ngcd == 1L)
+              build(num, lden * r.d)
+            else
+              build(div(num, ngcd), lden * div(r.d, ngcd))
```

Comment:
Sure, that would work too. But I can't imagine that this method would be useful from somewhere else, since it is very specific to this. So I would define it in LongRationals and make it private[this].